lib/events.js

```diff
@@ -307,3 +345,10 @@ function spliceOne(list, index) {
     list[i] = list[k];
   list.pop();
 }
+
+function arrayClone(arr, len) {
+  var arr_clone = new Array(len);
+  for (var i = 0; i < len; i += 1)
```

Comment:
maybe `arr_clone.push.apply(arr_clone, arr)`? might be a litter faster, haven't benchmarked it